source/uri-options/tests/single-threaded-options.json

```diff
@@ -0,0 +1,25 @@
+{
+  "tests": [
+    {
+      "description": "Valid options specific to single-threaded drivers are parsed correctly",
```

Comment:
Can we add a note to the `README.rst` about what to do about this test in a driver that is not multi-threaded? I'm guessing we just don't add this `.json` file to the test directory, but it might save someone a bit of pain down the line to be explicit about it.

Reply:
If multi-threaded clients do not recognize single-threaded options, shouldn't multi-threaded clients warn when such options are encountered? And therefore there should be multi-threaded client tests that expect these warnings?

Reply:
The connection string spec already specified that drivers warn on unrecognized options; for multithreaded options, this will just be like any other unrecognized option.

The README mentions that drivers need to skip optional tests for options they don't implement. In the case of this test, I think the cleanest solution is just for multi-threaded drivers not to include the file in the repo, although if a driver author prefers to skip it programmatically, that's fine too.

Reply:
I think this spec is explicitly not about how each driver should handle an option that it doesn't support, but just how it should handle certain ones that it does support. I'm not fond of the idea of introducing changes to the format of the connection string spec tests (which currently have no mechanism for differentiating between single- and multi-threaded drivers) and therefore requiring spec test runner changes; because the only tests that would require those sort of changes are not in the scope of the spec in the first place, I'd much rather give driver authors the ability to fix the problem non-programmatically.

Reply:
I assume this conversation was resolved, but I don't think it's a problem. Single-threaded drivers will likewise need to skip the connection pool test file.